### PR TITLE
Calispel: Proper concurrency in Nyxt at last!

### PR DESCRIPTION
**Pull request by Ambrevar**
I've replaced the ChanL library with Calispel which seems much, much better!

See https://github.com/CodyReichert/awesome-cl/issues/290#issuecomment-726032117 for a discussion.

I took the opportunity to (hopefully) fix the startup race condition.  @aartaka Can you make sure this branch starts properly for you?

Finally, I've made use of `fair-alt` instead of the interrupt-channel juggling business which was ugly and hard to understand.
This `fair-alt` will become essential in the new minibuffer library.

### CODE REVIEW

**Comment by aartaka** [comment]
This branch starts properly for me.

**Comment by Ambrevar** [comment]
Yay!

**Comment by jmercouris** [comment]
Branch appears to work for me too! woohoo!

First bug report: Reduce to buffer command will activate, but will not show the minibuffer. I have to cancel the command, try to reactivate it, and then sometimes it shows. I am not sure what the sequence of events are to get it work.

**Comment by Ambrevar** [comment]
It's the same issue with all chained minibuffers.

Try:

--8<---------------cut here---------------start------------->8---
- M-x open-file RET
--8<---------------cut here---------------end--------------->8---

The current workaround: Bind the command ;)

This issue is also on master, but the race condition happens way less
often.

It's actually good we have Calispel now so we can more reliable test if
we've fixed this issue properly.

My guess is that `hide` is run too late and hides the new minibuffer.
It's probably easy to fix this, I'll work on it later.

**Comment by jmercouris** [comment]
Ah, yes! I remember we had discussed a similar bug at some point in the recent past.

**Comment by jmercouris** [comment]
Perhaps it was the very same bug.

**Comment by Ambrevar** [comment]
To clarify, this race condition only occurs _frequently_ (about 3 times
out of 4 for me) with Calispel.

**Comment by Ambrevar** [comment]
Before marging:

- Fix aforementioned minibuffer issue.
- Package Calispel for Guix.
- Test for potential global dead lock when multiple threads are listening.
- Test Calispel on #1051.

**Comment by Ambrevar** [comment]
- [X] Global dead lock does not seem to be a problem.

- [X] I've just switched to Calispel in #1051.

**Comment by Ambrevar** [comment]
- [X] Minibuffer race condition should now be fixed.

File: source/browser.lisp [hunk 1]
```diff
@@ -215,12 +215,11 @@ editor executable."))
       (hooks:run-hook *after-init-hook*)
     (error (c)
       (log:error "In *after-init-hook*: ~a" c)))
-  (chanl:pexec ()
```
Comment (aartaka) [inline source/browser.lisp:218]: Another use of `chanl:pexec` is replaced with new `nyxt:pexec`. Why isn't it the case there? Should it be documented?

Comment (Ambrevar) [reply source/browser.lisp:218]: Sorry, I don't understand.  Where isn't it the case?  `nyxt:pexec` is just a dumb convenience wrapper around `bt:make-thread`.

Comment (Ambrevar) [reply source/browser.lisp:218]: Oh!  I see what you mean: this is `pexec` was the race condition you suffered from for weeks! :)  It was wrongly introduced during the sync minibuffer experiment and I had forgotten to remove it.

**Comment by Ambrevar** [comment]
- [X] Package Calispel for Guix.

All done.

Has anyone experienced any other issues with this patch?

**Comment by Ambrevar** [comment]
Any other review?  Did anyone have the chance to look at the code?
Let me know if this is OK to merge.

**Comment by Ambrevar** [comment]
@jmercouris: Can you test this on macOS?  Thanks!

**Comment by jmercouris** [comment]
Works on macOS! Merge when ready!

**Comment by Ambrevar** [comment]
Merged with c0b6b9e8be7536638196fef86d3aeaef3b342d1a
Thank you all for testing!

**Pull request closed without merging**